indra/newview/lltoolgrab.cpp

```diff
@@ -147,7 +147,7 @@ bool LLToolGrabBase::handleMouseDown(S32 x, S32 y, MASK mask)
     if (!gAgent.leftButtonGrabbed() || ((mask & DEFAULT_GRAB_MASK) != 0 && !gAgentCamera.cameraMouselook()))
     {
         // can grab transparent objects (how touch event propagates, scripters rely on this)
-        gViewerWindow->pickAsync(x, y, mask, pickCallback, /*bool pick_transparent*/ true);
+        gViewerWindow->pickAsync(x, y, mask, pickCallback, true, false, true);
```

Comment:
So, if left button not grabbed by script or not in mouselook. Isn't the issues specifically about mouselook? How changing this pick help with mouselook?

Reply:
This really helps :-)

The 7th argument of the pickAsync() method is `pick_unselectable`
Its value goes to the constructor of the class `LLPickInfo` as the argument `pick_surface_info` AKA `pick_uv_coords`
and is stored in the member field `mWantSurfaceInfo`

Probably this is the wrong order of the values sent to the constructor because the 8th argument is named `pick_unselectable` but it always receives `true`

Anyway, to fix the issue we should sent `true` to `pick_surface_info` to have true in `mWantSurfaceInfo`

Reply:
And why this isn't an issues outside of mouselook? Does it launch a different pickAsync?

Reply:
See the next variant

Reply:
Looks good.
Suggestion, cache gSavedSettings.getBOOL("SelectInvisibleObjects") while you are at it.

Reply:
> P.S. Still confused why it's mouselook specific, but fix makes sense.

In Mouselook mode we use `pickAsync`, but with no Mouselook mode we use `pickImmediate`
That's why this mistake in `pickAsync` doesn't take effect with no Mouselook mode